cdi/src/main/java/org/hibernate/validator/internal/cdi/ValidatorFactoryBean.java

```diff
@@ -51,7 +53,7 @@
 
 	public ValidatorFactoryBean(BeanManager beanManager, ValidationProviderHelper validationProviderHelper) {
 		this.beanManager = beanManager;
-		this.destructibleResources = newHashSet( 4 );
+		this.destructibleResources = newHashSet( 5 );
```

Comment:
I kinda regret we started doing this for sets and maps.

Unfortunately, initializing them with the number of intended elements will cause a resize. The reason being we don't take the load factor into account which is 0.75 by default. So a resize will happen after inserting the 3rd or 4th element. So in order to let it hold 5 elements, we'd have to initialize it with 8 (8 * 0.75 > 5). But it's all such micro optimization, that I think we could go for the default constructor equally well.